### DISCUSSION TITLE
安装飞书适配器后启动报错。

### DISCUSSION THREAD

**Discussion by sulaymanyf** [nonebot/discussions#92] [问答]
01-26 17:42:55 [WARNING] aiocache | ujson module not found, using json
01-26 17:42:55 [WARNING] aiocache | msgpack not installed, MsgPackSerializer unavailable
Traceback (most recent call last):
  File "D:\develop\python\upwork\bot\bot.py", line 5, in <module>
    from nonebot.adapters.feishu import Bot as FeishuBot
  File "C:\Users\my\.virtualenvs\upwork-Ib4vrkhY\lib\site-packages\nonebot\adapters\feishu\__init__.py", line 10, in <module>
    from .bot import Bot as Bot
  File "C:\Users\my\.virtualenvs\upwork-Ib4vrkhY\lib\site-packages\nonebot\adapters\feishu\bot.py", line 11, in <module>
    from nonebot.drivers import Driver, HTTPRequest, HTTPResponse
ImportError: cannot import name 'HTTPRequest' from 'nonebot.drivers' (C:\Users\my\.virtualenvs\upwork-Ib4vrkhY\lib\site-packages\nonebot\drivers\__init__.py)

**Comment by sulaymanyf** [comment]
是通过脚手架创建的项目

**Comment by yanyongyu** [comment]
请查看一下你的nonebot版本是否是alpha16

```
pip show nonebot2
```

**Reply by sulaymanyf** [reply]
不是 是 2.0.0.b1

**Reply by sulaymanyf** [reply]
[tool.poetry.dependencies]
python = "^3.7.3"
nonebot2 = "^2.0.0-beta.1"

[tool.poetry.dev-dependencies]
nb-cli = "^0.6.0"

[tool.nonebot]
plugins = []
plugin_dirs = ["src/plugins"]

[build-system]
requires = ["poetry_core>=1.0.0"]
build-backend = "poetry.core.masonry.api"

**Reply by sulaymanyf** [reply]
Version: 2.0.0b1
Summary: An asynchronous python bot framework.
Home-page: https://v2.nonebot.dev/
Author: yanyongyu
Author-email: yyy@nonebot.dev
License: MIT
Location: c:\users\yefan\.virtualenvs\upwork-ib4vrkhy\lib\site-packages
Requires: fastapi, loguru, pydantic, pygtrie, tomlkit, typing-extensions, uvicorn, yarl
Required-by: nb-cli, nonebot-adapter-feishu

**Reply by yanyongyu** [reply]
飞书尚未适配beta1版本哦，请使用0.5.3版本的nb-cli

卸载nonebot2以及nb-cli

```
pip install nb-cli==0.5.3
```

**Reply by sulaymanyf** [reply]
好的 谢谢

**Comment by StarHeartHunt** [comment]
飞书目前已经适配 nonebot 2.0.0-beta.1

**Reply by thomashooo** [reply]
> 飞书目前已经适配 nonebot 2.0.0-beta.1
请问一下，最新2.0.0rc2适配了飞书（lark）吗？
Name: nonebot2
Version: 2.0.0rc2 

顺带问一下，如何配置飞书机器人那边？

**Reply by thomashooo** [reply]
12-12 01:24:26 [INFO] uvicorn | 3.225.193.57:42856 - "POST /feishu/cli_a31ff89c0bb8900a HTTP/1.1" 200
12-12 01:24:26 [DEBUG] nonebot | FEISHU | check_at_me
12-12 01:24:26 [DEBUG] nonebot | FEISHU | message is 。hello
12-12 01:24:26 [SUCCESS] nonebot | feishu cli_a31ff89c0bb8900a | [message.p2p]: om_7bdc4974a3a796ad305d7dc5703c3893 from ou_6c81a71824667d7eabe662edd7e18a0f@[p2p:oc_26a85d4a539323ef054e976588e8e3f7] 。hello
12-12 01:24:26 [DEBUG] nonebot | Checking for matchers in priority 1...
12-12 01:24:26 [DEBUG] nonebot | Checking for matchers in priority 999...
12-12 01:24:26 [INFO] nonebot | Event will be handled by Matcher(type='message', module=nonebot_plugin_chatgpt.utils)
12-12 01:24:26 [DEBUG] nonebot | Running Matcher(type='message', module=nonebot_plugin_chatgpt.utils)
12-12 01:24:26 [DEBUG] nonebot | Running handler Dependent(call=ai_chat, parameterless=(Depends(Dependent(call=check_cooldown)),))
12-12 01:24:26 [DEBUG] nonebot | Handler Dependent(call=ai_chat, parameterless=(Depends(Dependent(call=check_cooldown)),)) skipped
12-12 01:24:26 [INFO] nonebot | Matcher(type='message', module=nonebot_plugin_chatgpt.utils) running complete
12-12 01:24:26 [DEBUG] nonebot | Stop event propagation

**Reply by StarHeartHunt** [reply]
A-kirami/nonebot-plugin-chatgpt 这个插件没有写对飞书的适配

**Reply by thomashooo** [reply]
1、感谢解惑，确实如此：https://github.com/A-kirami/nonebot-plugin-chatgpt/issues/39 插件只支持onebot v11方式。
2、这块有什么变通解决方案吗？这方面知识有限，有劳给个思路，如何下手适配（比如是不是有方式把feishu转发到v11？还是必须去写一套chatgpt的飞书版？）

**Reply by StarHeartHunt** [reply]
> 1、感谢解惑，确实如此：[A-kirami/nonebot-plugin-chatgpt#39](https://github.com/A-kirami/nonebot-plugin-chatgpt/issues/39) 插件只支持onebot v11方式。 2、这块有什么变通解决方案吗？这方面知识有限，有劳给个思路，如何下手适配（比如是不是有方式把feishu转发到v11？还是必须去写一套chatgpt的飞书版？）

需要再写一套飞书平台的逻辑。

但是因为他是纯文本类型的插件，不需要太多的改动。只需要将一些事件类型标注改为飞书和 OneBot 平台的集合，或者直接使用基类。具体可以参考适配器文档和 NoneBot 2 文档。

**Comment by kunhn** [comment]
今天使用脚手架安装飞书适配器，飞书后台配置事件请求地址的时候提示  响应 返回的不是json格式，是不是也和适配度有关，
我的nb2 版本如下：
Name: nonebot2
Version: 2.0.0rc2
Summary: An asynchronous python bot framework.
Home-page: https://v2.nonebot.dev/
Author: yanyongyu
Author-email: yyy@nonebot.dev
License: MIT
Location: /Users/apple/opt/anaconda3/envs/nba/lib/python3.8/site-packages
Requires: fastapi, loguru, pydantic, pygtrie, tomlkit, typing-extensions, uvicorn, yarl


另外请问飞书目前适配的nb2版本是哪个版本。
可否厚脸皮的请求一下哈哈
如果有可能，以后在更新后，是否可以写在飞书适配器的说明文档里面

**Reply by StarHeartHunt** [reply]
1. 飞书适配器的版本号与 NoneBot 2 无关，目前飞书适配器 2.0.0-beta.6 版本依赖的就是 2.0.0rc2
2. 看 log 我不清楚你是不是在有公网 IP 或者做了内网穿透的情况下启动的 NoneBot。

**Reply by thomashooo** [reply]
你自己先外网：curl  你的url看响应是不是json：http://地址:端口/feishu/你的appid

**Comment by kunhn** [comment]
目前nb2的版本有这些  (from versions: 2.0.0a1, 2.0.0a2, 2.0.0a3, 2.0.0a4, 2.0.0a5, 2.0.0a6, 2.0.0a7, 2.0.0a8, 2.0.0a8.post1, 2.0.0a8.post2, 2.0.0a9, 2.0.0a9.post1, 2.0.0a10, 2.0.0a11, 2.0.0a12, 2.0.0a13, 2.0.0a13.post1, 2.0.0a14, 2.0.0a15, 2.0.0a16, 2.0.0b1, 2.0.0b2, 2.0.0b3, 2.0.0b4, 2.0.0b5, 2.0.0rc1, 2.0.0rc2)

我看作者大大的飞书适配器版本最新到b6，但是我看官方只有b5
是不是飞书的b5/b6对应的都是nb2的b5版本

**Reply by kunhn** [reply]
晚上我有使用了nb2.0.0b3和飞书的2.0.0b3尝试，这次的请求地址配置提示结果是url invalid无效地址，但是我在nb里面都有配置端口的，

是不是服务器没有接上我的本地ip，所以导致我用公网ip接端口无效
终端返回如下
12-13 01:51:14 [SUCCESS] nonebot | NoneBot is initializing...
12-13 01:51:14 [INFO] nonebot | Current Env: dev
12-13 01:51:14 [WARNING] __main__ | Always use `nb run` to start the bot instead of manually running!
12-13 01:51:14 [SUCCESS] nonebot | Running NoneBot...
12-13 01:51:14 [INFO] uvicorn | Started server process [24695]
12-13 01:51:14 [INFO] uvicorn | Waiting for application startup.
12-13 01:51:14 [INFO] nonebot | FEISHU | Bot cli_a391c8c8a3b8900c connected
12-13 01:51:14 [INFO] uvicorn | Application startup complete.
12-13 01:51:14 [INFO] uvicorn | Uvicorn running on http://0.0.0.0:3000 (Press CTRL+C to quit)
^C12-13 01:56:51 [INFO] uvicorn | Shutting down
12-13 01:56:51 [INFO] uvicorn | Waiting for application shutdown.
12-13 01:56:51 [INFO] uvicorn | Application shutdown complete.
12-13 01:56:51 [INFO] uvicorn | Finished server process [24695]
(nba) apple@kundeiMac feishubot_1 % /Users/apple/opt/anaconda3/envs/nba/bin/python /Users/apple/feishubot_1/bot.py
12-13 01:57:00 [SUCCESS] nonebot | NoneBot is initializing...
12-13 01:57:00 [INFO] nonebot | Current Env: dev
12-13 01:57:00 [WARNING] __main__ | Always use `nb run` to start the bot instead of manually running!
12-13 01:57:00 [SUCCESS] nonebot | Running NoneBot...
12-13 01:57:00 [INFO] uvicorn | Started server process [24801]
12-13 01:57:00 [INFO] uvicorn | Waiting for application startup.
12-13 01:57:01 [INFO] nonebot | FEISHU | Bot cli_a391c8c8a3b8900c connected
12-13 01:57:01 [INFO] uvicorn | Application startup complete.
12-13 01:57:01 [INFO] uvicorn | Uvicorn running on http://0.0.0.0:3000 (Press CTRL+C to quit)

请问我这两次的问题是同一个问题吗？问题出在哪里？应该怎么解决？
有劳大佬解答-。-

**Comment by kunhn** [comment]
是有公网IP的，所以很纳闷，不知道问题出在哪里，你那儿有跑通的源码吗，我参考一下发自我的手机-------- 原始邮件 --------发件人： StarHeart ***@***.***>日期： 2022年12月13日周二 04:22收件人： nonebot/discussions ***@***.***>抄送： kunhn ***@***.***>, Comment ***@***.***>主    题： Re: [nonebot/discussions] 安装飞书适配器后启动报错。 (Discussion #92)

飞书适配器的版本号与 NoneBot 2 无关，目前飞书适配器 2.0.0-beta.6 版本依赖的就是 2.0.0rc2看 log 我不清楚你是不是在有公网 IP 或者做了内网穿透的情况下启动的 NoneBot。

**Reply by thomashooo** [reply]
成功了吗？我还在等你一起研究，接入chatGPT呢？

**Reply by kunhn** [reply]
我已经试了好几个版本了，还是无法配置事件订阅，端口和IP是肯定没有问题的，我之前没有用nonebot的时候就可以正常配置和接受消息，朋友介绍说这个装插件很方便，就想转过来试试，但是一直配不上，就很郁闷。进入死胡同了，靠自己琢磨有点恼火

**Reply by kunhn** [reply]
加个飞书吧，方便交流
阿坤 邀请你加入飞书群，快点击https://applink.feishu.cn/client/chat/chatter/add_by_link?link_token=240s5807-0a5f-4fd6-9f91-258486a9bb23加入吧！

**Reply by kunhn** [reply]
飞群二维码飞书20221213-202837.png
https://www.aliyundrive.com/s/zVYKi6kMFVo
点击链接保存，或者复制本段内容，打开「阿里云盘」APP

**Reply by kunhn** [reply]
飞群二维码飞书20221213-202837.png
https://www.aliyundrive.com/s/zVYKi6kMFVo
点击链接保存，或者复制本段内容，打开「阿里云盘」APP

**Reply by kunhn** [reply]
成功了，已经可以正常聊天，欢迎大家一起来飞书群讨论

**Reply by Mautu** [reply]
来个群 进群学习学习
